pkg/kubectl/rollout_status.go

```diff
@@ -77,3 +83,23 @@ func (s *DeploymentStatusViewer) Status(namespace, name string, revision int64)
 	}
 	return fmt.Sprintf("Waiting for deployment spec update to be observed...\n"), false, nil
 }
+
+// Status returns a message describing daemon set status, and a bool value indicating if the status is considered done
+func (s *DaemonSetStatusViewer) Status(namespace, name string, revision int64) (string, bool, error) {
+	//ignoring revision as DaemonSets does not have history yet
+
+	daemon, err := s.c.DaemonSets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", false, err
+	}
+	if daemon.Generation <= daemon.Status.ObservedGeneration {
+		if daemon.Status.UpdatedNumberScheduled < daemon.Status.DesiredNumberScheduled {
+			return fmt.Sprintf("Waiting for rollout to finish: %d out of %d new pod have been updated...\n", daemon.Status.UpdatedNumberScheduled, daemon.Status.DesiredNumberScheduled), false, nil
+		}
+		if daemon.Status.NumberAvailable < daemon.Status.DesiredNumberScheduled {
```

Comment:
This should be `ds.Status.NumberAvailable < ds.Status.DesiredNumberScheduled-MaxUnavailable`